### PR TITLE
Update mono-interp to handle the same Vector.As APIs as mono-jit

### PR DESCRIPTION
**Pull request by tannergooding**
This is a simplified version of https://github.com/dotnet/runtime/pull/104049, trying to handle the smaller set of existing scenarios to start.

### CODE REVIEW

**Comment by tannergooding** [comment]
CC. @kg this should be ready for review

**Comment by kg** [comment]
lgtm, but i'd feel better if vlad or milos also looked at it. if the regressions this is fixing don't go away, i can update the jiterpreter.

**Comment by tannergooding** [comment]
> if the regressions this is fixing don't go away, i can update the jiterpreter.

Sounds good! This should help some, but there's some others that still need to be handled as well. I was having troubles handling them all at once in https://github.com/dotnet/runtime/pull/104049, so I decided to break it apart into smaller chunks instead starting with this one.

I'm going to get `AsVector`, `AsQuaternion`, `AsPlane`, `AsVector4`, and `AsVector128(Vector4)` next since those are also `same size bitcasts` and should be straightforward. Then that will leave `AsVector128(Vector2)`, `AsVector128(Vector3)`, and `AsVector128Unsafe` after that as the ones that change sizes and I believe were the ones causing the most issues in my previous attempt.

-- Could you give a brief explanation on the difference between the handling here and that in the jiterpreter? Does the logic not get shared between them for the SIMD acceleration?

**Comment by kg** [comment]
> > if the regressions this is fixing don't go away, i can update the jiterpreter.
> 
> Sounds good! This should help some, but there's some others that still need to be handled as well. I was having troubles handling them all at once in #104049, so I decided to break it apart into smaller chunks instead starting with this one.
> 
> I'm going to get `AsVector`, `AsQuaternion`, `AsPlane`, `AsVector4`, and `AsVector128(Vector4)` next since those are also `same size bitcasts` and should be straightforward. Then that will leave `AsVector128(Vector2)`, `AsVector128(Vector3)`, and `AsVector128Unsafe` after that as the ones that change sizes and I believe were the ones causing the most issues in my previous attempt.
> 
> -- Could you give a brief explanation on the difference between the handling here and that in the jiterpreter? Does the logic not get shared between them for the SIMD acceleration?

The jiterpreter has a generic path for interpreter SIMD opcodes that dispatches directly to the C helper, but with a wasm opcode mapping or dedicated implementation (i think this would need the latter), it can generate native jitcode directly. For this bitcast I think we'd just emit a single vector memory move.

**Comment by tannergooding** [comment]
@BrzVlad, @kotlarmilos, this should be ready for review

File: src/mono/mono/mini/interp/transform-simd.c
Comment (kotlarmilos) [inline src/mono/mono/mini/interp/transform-simd.c:334]: Should we also check the namespace?

**Review by kotlarmilos — APPROVED** [review]
LGTM! @BrzVlad Please sign-off before merge.

File: src/mono/mono/mini/interp/transform-simd.c
Comment (tannergooding) [reply src/mono/mono/mini/interp/transform-simd.c:334]: We aren't doing so in `simd-intrinsics.c`, notably, but we could if desired.

I believe the expectation is that we're already only hitting this path for `System.Runtime.Intrinsics` or `System.Numerics` due to the higher level functions that can dispatch to these code paths.

**Review by BrzVlad — APPROVED** [review]

**Review by fanyang-mono — APPROVED** [review]

**Comment by tannergooding** [comment]
Failure is the general timeout for this leg that is already tracked.

**Pull request merged**
